File: Software Development Module/Week12/PPSApp.java
/*
PPSApp.java
Billy Cussen
22/07/2020
*/

import java.util.Scanner;

public class PPSApp {
    
    public static void main(String[]args){

        //Variable
        String pps = new String();
        String repeat = new String();
        boolean inputCheck;
        //Object
        Scanner keyboard;
        PPS myPPS;

        do{
            do{
                keyboard = new Scanner(System.in);
                inputCheck = true;

                try{
                    System.out.print("Please enter your PPS Number:");
                    pps = keyboard.nextLine();

                    if(pps.length()==0){
                        throw new Exception();
                    }

                }catch(Exception e){
                    inputCheck = false;
                    System.out.println("Please enter your PPS Number correctly (ie. Not Blank)");
                }finally{
                    keyboard = null;
                }

            }while(!inputCheck);

            myPPS = new PPS(pps);
            System.out.println(myPPS.computeAndGetVaidity());
            myPPS = null;

            do{
                keyboard = new Scanner(System.in);
                System.out.print("Please enter Y or N to retry: ");
                repeat = keyboard.nextLine();

                if(!(repeat.equalsIgnoreCase("Y")||repeat.equalsIgnoreCase("N"))){
                    System.out.println("You need to enter Y or N");
                }

                keyboard = null;

            }while(!(repeat.equalsIgnoreCase("Y")||repeat.equalsIgnoreCase("N")));
            

        }while(repeat.equalsIgnoreCase("Y"));

    }

}

File: Python Course/Software Development Module - in Python/Week12/PPSApp.py
"""
PPSApp.py
Billy Cussen
17/10/2020
"""

#Objects
import gc
import PPS 
from PPS import myPPS

#Variables
retry = "y"


while retry == "y":

    inputCheck = True

    #Input
    while inputCheck:
        try:
            pps = str(input("Please enter your PPS Number: "))
            if(pps == ""):
                raise Exception("You can't enter anything blank, please enter your 8 Character PPS Number")
        except Exception as e:
            print(str(e))
        else:
            inputCheck = False

    myPPSApp = myPPS(pps)
    print(myPPSApp.checkPPS())
    myPPSApp = None
    gc.collect

    retry = ""
    while retry != "y" and retry != "n":
        retry = str(input("Please enter \"Y\" to retry or\"N\" to exit: "))
        retry = retry.lower()
        if retry != "y" and retry != "n":
            print("You need to enter \"Y\" or \"N\" (any case)")

print("Thanks for using my PPS App")